lib/matplotlib/text.py

```diff
@@ -279,6 +279,38 @@ def _get_multialignment(self):
         else:
             return self._horizontalalignment
 
+    def _char_index_at(self, x):
+        """
+        Calculates the index nearer to x, we accumulate
+        the width of each character contained in the text
+        and find the index at which the distance between the
+        click position and the accumulated sum is minimal.
+        This works only on single line texts.
+        """
+        if not self._text:
+            return 0
+
+        if (not hasattr(self, "_saved_fontproperties") or
+                str(self._fontproperties) != self._saved_fontproperties):
+
+            self._saved_fontproperties = str(self._fontproperties)
+            self._chars_size = dict()  # to cache results
```

Comment:
Thinking about this a bit more: Why are we having the char size cache on the instance? We could instead have a class wide cache `Text._charsize_cache[self._fontproperties][char]`. Note that FontProperties are hashable and thus can serve as keys directly.

Reply:
mmmh yes this seems a good idea but we need, I think, to use anyway `str(self._fontproperties)` for keys because font property remain the same object when you do something like `self.set_fontsize(100)`

Reply:
Fair point. Let me think about this - I don't have time right now, but will look into it soon.

Reply:
No problem, anyway using the string doesn't seem so bad to me

Reply:
@timhoffm how you think it is now?